netsim/extra/firewall.zonebased/plugin.py

```diff
@@ -0,0 +1,43 @@
+# Firewall Zone-Based Plugin
+## Simple rules for now, only default policy supported initially
+
+from netsim.utils import log
+from netsim.augment import addressing, devices
+from netsim import api,data
+from box import Box
+import netaddr
+
+_config_name = 'firewall.zonebased'
+
+def post_transform(topology: Box) -> None:
+    global _config_name
+    for node in topology.nodes.values():
+        features = devices.get_device_features(node,topology.defaults)
+        zbf_supported = 'firewall.zonebased' in features
+        if zbf_supported:
+            # Create basic structures if not exists
+            if 'firewall' not in node:
+                node.firewall = {}
+            if 'firewall.zonebased' not in node:
```

Comment:
I would turn this check around -- if ZBF is not supported, but firewall.zonebased exists, we would report an error.